src/sage_setup/cython_options.py

```diff
@@ -10,13 +10,17 @@ def compiler_directives(profile: bool):
         auto_pickle=False,
         # Do not create __test__ dictionary automatically from docstrings
         autotestdict=False,
+        binding=False,
+        c_api_binop_methods=True,
```

Comment:
Why `binding=False`?

`binding=True` is a present that we were expecting from cython3 for a long time. #26254

Reply:
I think the goal of this PR is to be able to build sagemath with cython 3. It seems enabling options to make cython 3 keep legacy behaviour is the minimal working step we can make. After sagemath is built with cython 3, it can be explored how to switch to the new behaviour. Same for `legacy_implicit_noexcept`, which can be removed after all the implicit `noexcept` are made explicit, etc.

Reply:
Fair enough.